### PR TITLE
ASoC: SOF: ipc3-topology: Fixes for bytes control loading, handling

### PR DESCRIPTION
**Pull request by ujfalusi**
Hi,

the first patch is picked from upstream as the issue was reported there:: https://lore.kernel.org/alsa-devel/20220427185221.28928-1-peter.ujfalusi@linux.intel.com/

The second patch is correcting other issue, not reported upstream, but caught my eyes, we can send it after the internal review.

### CODE REVIEW

File: sound/soc/sof/ipc3-topology.c [hunk 1]
```diff
@@ -1593,24 +1593,23 @@ static int sof_ipc3_control_load_bytes(struct snd_sof_dev *sdev, struct snd_sof_
 	struct sof_ipc_ctrl_data *cdata;
```
Comment (ujfalusi) [inline sound/soc/sof/ipc3-topology.c:1593]: Missing fixes tag:
Fixes: b5cee8feb1d48 ("ASoC: SOF: topology: Make control parsing IPC agnostic")

File: sound/soc/sof/ipc3-topology.c [hunk 2]
```diff
@@ -23,7 +23,8 @@
 struct sof_widget_data {
```
Comment (ujfalusi) [inline sound/soc/sof/ipc3-topology.c]: I could have added Sergey's Reviewed and Tested -by tags to this.

File: sound/soc/sof/ipc3-topology.c (hunk 1 above)
Comment (plbossart) [reply sound/soc/sof/ipc3-topology.c:1593]: @ujfalusi are you planning to update this branch?

Comment (ujfalusi) [reply sound/soc/sof/ipc3-topology.c:1593]: Oh, OK, I'll update with the Fixes tag. @ranj063, can you take a look if I got this right?

**Comment by ujfalusi** [comment]
Changes since v1:
- First patch is cherry-picked from upstream
- Fixes tag added to patch 2

**Review by plbossart — DISMISSED** [review]

**Comment by plbossart** [comment]
@ujfalusi this seems to conflict?

**Comment by ujfalusi** [comment]
> @ujfalusi this seems to conflict?

Oh, The first patch comes back from upstream, just a sec...

**Comment by ujfalusi** [comment]
Changes since v2:
- drop first patch as it comes from upstream now

**Review by plbossart — APPROVED** [review]

**Comment by plbossart** [comment]
@ranj063 @bardliao @juimonen Can you please review?

**Review by ranj063 — APPROVED** [review]

**Review by bardliao — APPROVED** [review]

**Pull request merged**